src/ssort/_requirements.py

```diff
@@ -52,12 +53,22 @@ def _get_requirements_for_function_def(
     for decorator in node.decorator_list:
         yield from get_requirements(decorator)
 
-    yield from get_requirements(node.args)
+    scope: set[str] = set()
+    if sys.version_info >= (3, 12):
+        for type_param in node.type_params:
+            yield from get_requirements(type_param)
```

Comment:
The PEP requires that later params are evaluated with earlier variables in scope.  I think this needs to be filtered to remove bindings by earlier params.